test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java

```diff
@@ -4086,6 +4236,25 @@ public MockUserGroupSoap(Users users) {
       return requestContext;
     }   
   }
+  
+  private static class MockUserGroupSoapException
+      extends UnsupportedUserGroupSoap {
+    private Map<String, Object> requestContext = new HashMap<String, Object>();
+    // counter for site user mapping call
+    private AtomicLong atomicNumberGetSiteUserMappingCalls = new AtomicLong(0);
+
+    @Override
+    public GetUserCollectionFromSiteResponse.GetUserCollectionFromSiteResult
+        getUserCollectionFromSite() {
+          atomicNumberGetSiteUserMappingCalls.incrementAndGet();
+          throw new SOAPFaultException(null);
```

Comment:
This isn't actually doing what it looks like it is. This line throws a NPE because the constructor arg is null, and the parameter is a SOAPFault, which is hard to construct. We could mock one, but the other code in this class is using WebServiceException instead (a superclass of SOAPFaultException). Let's change this to:

throw new WebServiceException("Mock SOAP error");

This is the same code path as a NPE (or a SOAPFaultException), and this seems more clear about the intent.